code/game/objects/explosion.dm

```diff
@@ -95,6 +95,27 @@
 		for (var/turf/T in trange(max_range, epicenter))
 			var/dist = cheap_pythag(T.x - x0, T.y - y0)
 
+			if(explosion_newmethod)	//realistic explosions that take obstacles into account
+				var/turf/Trajectory = T

```

Comment:
In the code as it is right now yes, but I was sleep deprived when I coded that, I'm rewriting it so it's less "hardcoded" as Comic is pointing out. Turfs and Doors will now have an explosion_block var, so I need to keep Trajectory defined as turf.